apps/CardinalBanditsPureExploration/algs/EpsilonGreedy.py

```diff
@@ -0,0 +1,113 @@
+"""
+LilUCB app implements CardinalBanditsPureExplorationPrototype
+author: Kevin Jamieson
+last updated: 11/13/2015
+"""
+
+import numpy
+import numpy.random
+import next.utils as utils
+
+class MyAlg:
+
+  def initExp(self,butler,n,R,failure_probability,params={}):
+    butler.algorithms.set(key='n', value=n)
+    butler.algorithms.set(key='delta', value=failure_probability)
+    butler.algorithms.set(key='R', value=R)
+    butler.algorithms.set(key='pT', value=0)
+
+    empty_list = numpy.zeros(n).tolist()
+    butler.algorithms.set(key='Xsum',value=empty_list)
+    butler.algorithms.set(key='X2sum',value=empty_list)
+    butler.algorithms.set(key='T',value=empty_list)
+
+    priority_list = numpy.random.permutation(n).tolist()
+    butler.algorithms.set(key='priority_list',value=priority_list)
+
+    return True
+
+
+  def getQuery(self,butler,participant_uid):
+
+    # Get Query Indicies
+    kv_dict = butler.algorithms.increment_many(key_value_dict={'priority_list':0,'priority_list_cnt':1})
+    priority_list = kv_dict['priority_list'] # List of queries?
+    priority_list_cnt = kv_dict['priority_list_cnt'] # Length of queries?
+
+    # EpsilonGreedy Implementation
+    key_value_dict = butler.algorithms.get()
+    t = key_value_dict['pT']
+    pt = 1 - (1 / (t + 1))
+    if numpy.random.rand() < pt:
+        index = priority_list[0]
+    index = numpy.random.choice(priority_list)
+
+    # Update t for next run
+    newT = t + 1
+    butler.algorithms.set(key='pt', value=newT)
```

Comment:
I think there's a typo here (`pt` vs `pT`). But maybe we could provide a longer name, maybe `answers_received`? That'd be a little easier to maintain.

Reply:
I agree, a more descriptive name would be better.